### DISCUSSION TITLE
找不到TCGA数据集Silde Image类型数据标签

### DISCUSSION THREAD

**Discussion by RuixiangZhao** [ShixiangWang/self-study#92] [Q&A] [answered]
各位大佬好，最近开始入门病理切片分析，要复现的一个工作用到了`TCGA-LUAD`数据集，想象大家请教两个有关TCGA数据集的问题：
+ TCGA中Diagnostic Slide类型的数据的标签在哪里？是`Annotations`那一列吗？如果是的话，我看这一列出现了0,1,2,3，那意思是0代表无病，1,2,3分别代表三种疾病？
<img width="702" alt="1" src="https://user-images.githubusercontent.com/62002839/145774971-0bb0b622-2baa-4925-9b1a-b813470999fd.PNG">

+ 还有一个关于TCGA数据命名问的问题，我看他[官方文档](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/)介绍的命名规则（如下图）为什么和实际的文件名（可以看上面图片的文件名）不一致？（我本来想试着从它的文件名中找到标签信息
![barcode](https://user-images.githubusercontent.com/62002839/145775332-afcd2df0-6c4a-4365-8a36-3abc323f36bf.png)

**Comment by ShixiangWang** [comment, marked as answer]
第一个问题没有接触过，所以没有什么建议。

第二个需要区别sample barcode和file name。第一个截图里的file name是说明文件的一个编号，而第二个截图说的是肿瘤（或正常）样本的编号。比如同一个样本，可能做了2个不同的实验，产生了一堆不同的数据结果文件。

第二个问题的解决可以将所要分析的文件添加到购物车，然后点击购物车，下载相关的元数据文件。我记得有个sample sheet非常重要，里面记录了文件和样本的关系。另外，GDC还采用了UUID的编码体系。http://www.bioconductor.org/packages/release/bioc/html/TCGAutils.html 这个包可能会有用。

**Reply by RuixiangZhao** [reply]
@ShixiangWang 非常感谢您这么快的回复！
在你的建议下，我又仔细看了一遍这个数据集对应的`metadata.json`文件，发现`Annotations`这一列的含义并非是疾病标签，而是代表着对某张图片**注释的个数**：
比如一张`Annotations=0`的slide image在`metadata.json`对应的信息如下（字典里没有"annotations"这一关键词）：
<details>
<pre>
{
  "data_format": "SVS", 
  "access": "open", 
  "associated_entities": [
    {
      "entity_submitter_id": "TCGA-44-6777-01Z-00-DX1", 
      "entity_type": "slide", 
      "case_id": "349e6f38-2c67-4a69-a777-b9173a2a27cf", 
      "entity_id": "67dd4a3a-2acf-4f54-9f98-ad6e19318bbf"
    }
  ], 
  "file_name": "TCGA-44-6777-01Z-00-DX1.aca7b23d-a601-4476-8bfd-eb46f0b6a96e.svs", 
  "md5sum": "9ea1398060a4960d17f4b09bb3913f63", 
  "file_id": "cba0303e-5f68-433c-8459-a3565bc7ccab", 
  "data_type": "Slide Image", 
  "submitter_id": "TCGA-44-6777-01Z-00-DX1_slide_image", 
  "data_category": "Biospecimen", 
  "state": "released", 
  "experimental_strategy": "Diagnostic Slide", 
  "file_size": 304926153
}
</pre>
</details>

而一张`Annotations=2`的slide image在`metadata.json`对应的信息中"annotations"属性列表长度为2：
<details>
<pre>
{
  "data_format": "SVS", 
  "access": "open", 
  "associated_entities": [
    {
      "entity_submitter_id": "TCGA-93-8067-01Z-00-DX1", 
      "entity_type": "slide", 
      "case_id": "bbe88801-34f3-46d2-bbfd-b46c3901ed71", 
      "entity_id": "7f1dfe61-3972-4650-a4ed-919f531f5dd3"
    }
  ], 
  "file_name": "TCGA-93-8067-01Z-00-DX1.325d0cd5-9bb5-4f51-a197-75d3c3d19aaf.svs", 
  "annotations": [
    {
      "entity_submitter_id": "TCGA-93-8067", 
      "notes": "Prior malignancy  prostate cancer treated with locoregional radiation.", 
      "submitter_id": "11972", 
      "entity_id": "bbe88801-34f3-46d2-bbfd-b46c3901ed71", 
      "classification": "Notification", 
      "created_datetime": "2012-11-10T00:00:00", 
      "annotation_id": "6a005f5b-af6c-5584-80ac-4d1baff2535a", 
      "entity_type": "case", 
      "updated_datetime": "2018-08-23T16:32:20.747393-05:00", 
      "case_id": "bbe88801-34f3-46d2-bbfd-b46c3901ed71", 
      "state": "released", 
      "category": "History of acceptable prior treatment related to a prior/other malignancy", 
      "status": "Approved"
    }, 
    {
      "entity_submitter_id": "TCGA-93-8067", 
      "notes": "Prior malignancy  prostate cancer treated with locoregional radiation.", 
      "submitter_id": "11675", 
      "entity_id": "bbe88801-34f3-46d2-bbfd-b46c3901ed71", 
      "classification": "Notification", 
      "created_datetime": "2012-10-31T00:00:00", 
      "annotation_id": "0e7aa771-e53e-5678-b4e2-d96cfd9446b2", 
      "entity_type": "case", 
      "updated_datetime": "2018-08-23T16:32:20.747393-05:00", 
      "case_id": "bbe88801-34f3-46d2-bbfd-b46c3901ed71", 
      "state": "released", 
      "category": "Prior malignancy", 
      "status": "Approved"
    }
  ], 
  "submitter_id": "TCGA-93-8067-01Z-00-DX1_slide_image", 
  "data_category": "Biospecimen", 
  "file_size": 1312234591, 
  "md5sum": "9b8591597a92c83e63919b302c61f2d5", 
  "file_id": "09acdcf2-f634-452f-b78b-975128e7d14b", 
  "data_type": "Slide Image", 
  "state": "released", 
  "experimental_strategy": "Diagnostic Slide"
}
</pre>
</details>

但我仍未找到一张Slide Image对应的标签在哪里，如果有人知道的话，麻烦您告诉我一下，多谢！

**Reply by ShixiangWang** [reply]
我把你的问题发到生信技能树讨论了下。你看看最新的release数据中能不能找到有用的信息 https://docs.gdc.cancer.gov/Data/Release_Notes/Data_Release_Notes/#data-release-310。

**Reply by RuixiangZhao** [reply]
我是在11月底下的数据，因此是在31.0版本之后的最新数据，而且我看了它的版本更新内容，感觉和标签信息没关系。

不过在您第一个回答的启发下，我发现标签信息仍然隐藏在文件名之中。正如您上面所说，每个病例会采集多个样本，每个样本有一个barcode，如下图：
<img width="154" alt="1" src="https://user-images.githubusercontent.com/62002839/145812049-6a32e496-6753-4945-b179-60e6e2b18865.PNG">
标签信息就隐藏在这个barcode之中，而文件名的前缀会和这个barcode保持一致，因此也通过文件名也可以找到文件的标签信息，规则见[官方文档](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/)

总之，非常感谢你的帮助！
（ps：顺便吐槽一下它网站上`Annotations`这一列信息，太容易误导人了

**Reply by kdzkdz** [reply]
这个数据集有yolo的标签吗

**Reply by kdzkdz** [reply]
> 我是在11月底下的数据，因此是在31.0版本之后的最新数据，而且我看了它的版本更新内容，感觉和标签信息没关系。
> 
> 不过在您第一个回答的启发下，我发现标签信息仍然隐藏在文件名之中。正如您上面所说，每个病例会采集多个样本，每个样本有一个barcode，如下图： <img alt="1" width="154" src="https://user-images.githubusercontent.com/62002839/145812049-6a32e496-6753-4945-b179-60e6e2b18865.PNG"> 标签信息就隐藏在这个barcode之中，而文件名的前缀会和这个barcode保持一致，因此也通过文件名也可以找到文件的标签信息，规则见[官方文档](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/)
> 
> 总之，非常感谢你的帮助！ （ps：顺便吐槽一下它网站上`Annotations`这一列信息，太容易误导人了

你好，这个数据集有yolo的格式吗

**Reply by ShixiangWang** [reply]
> > 我是在11月底下的数据，因此是在31.0版本之后的最新数据，而且我看了它的版本更新内容，感觉和标签信息没关系。
> > 不过在您第一个回答的启发下，我发现标签信息仍然隐藏在文件名之中。正如您上面所说，每个病例会采集多个样本，每个样本有一个barcode，如下图： <img alt="1" width="154" src="https://user-images.githubusercontent.com/62002839/145812049-6a32e496-6753-4945-b179-60e6e2b18865.PNG"> 标签信息就隐藏在这个barcode之中，而文件名的前缀会和这个barcode保持一致，因此也通过文件名也可以找到文件的标签信息，规则见[官方文档](https://docs.gdc.cancer.gov/Encyclopedia/pages/TCGA_Barcode/)
> > 总之，非常感谢你的帮助！ （ps：顺便吐槽一下它网站上`Annotations`这一列信息，太容易误导人了
> 
> 你好，这个数据集有yolo的格式吗

一般是 svs

**Reply by Tengxf** [reply]
> ####

Holle,your answer is useful.How can I get barcode?Wish you have a good day!

**Comment by Carrollkx** [comment]
你好，问题解决了吗？我最近也在研究WSI相关的问题，想问一下TCGA下载的WSI数据的标签在哪里看

**Comment by lhk0504** [comment]
楼主能找到Diagnostic Slide数据的标注信息吗

**Reply by ShixiangWang** [reply]
不熟悉
